packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -349,7 +254,7 @@ export class HttpApi extends HttpApiBase {
   /**
    * default stage of the api resource
    */
-  public readonly defaultStage: HttpStage | undefined;
+  public readonly defaultStage: IStage | undefined;
```

Comment:
Similar comment to another one. I think there's value in going from IHttpApi -> IHttpStage.
If not, customers will have to do type checking to correctly get to `IHttpStage`.

Perhaps we should move this down to the subclasses. I'm curious what the implications of that would be.

Reply:
Moving this down to subclasses will not be possible because of how `ApiMapping` works with the new implementation. It requires stage to be present for it's functionalities. So either we create a specific `ApiMapping` per API type (like we had before) or we keep it like so.

Reply:
Could we do something like this instead?

```ts
public get defaultStage(): IHttpStage | undefined {
  if (_defaultStage && !(_defaultStage instanceof IHttpStage)) {
    throw new Error('unexpected error: defaultStage is not of type IHttpStage');
  }
  return _defaultStage;
}
```

The error condition should never occur unless there's a bug within the CDK.

Reply:
Actually, can you explain your previous comment? I can't quite figure out, from the code diffs, why this is necessary for `ApiMapping`.

Reply:
So the `ApiMapping` requires `api` to be passed as props. This `api` with the new implementation takes the form of `IApi` since it is common for both API types. In the implementation of `ApiMapping` we use `defaultStage` if a `stage` is not provided in the props. Now, if we move the `defaultStage` declaration from `IApi` down to individual subclasses, we will not be able to do this. Let me know if that was clear. Also, I am not sure how the above recommendation would pan out. Do we declare this getter in `IApi` too?

Reply:
Given that default stage is a concept that only (seems to) exist on Http APIs, we should bring this down to the HttpApi subclass. Specifically, the logic of falling back to the default stage if a stage is not provided should be specific to `HttpApi`.

Reply:
Changed the implementation in a non breaking way for `ApiMapping` (although the rename from `HttpApiMapping` would still be a breaking change)